DatadogSessionReplay/Sources/Writers/ResourcesWriter.swift

```diff
@@ -15,26 +15,106 @@ internal protocol ResourcesWriting {
+                        switch result {
+                        case .value(let data, _):
+                            do {
+                                if let knownIdentifiers = try data.asKnownIdentifiers() {
+                                    self?.knownIdentifiers.formUnion(knownIdentifiers)
+                                }
+                            } catch let error {
+                                self?.scope.telemetry.error("Failed to decode known identifiers", error: error)
+                            }
+                        default:
+                            break
+                        }
+                    }
+                } else { // Reset if store was created more than 30 days ago
+                    self?.scope.dataStore.setValue(
+                        Date().timeIntervalSince1970.asData(),
+                        forKey: Constants.storeCreationKey
+                    )
+                    self?.scope.dataStore.removeValue(forKey: Constants.knownResourcesKey)
+                }
+            } catch let error {
+                self?.scope.telemetry.error("Failed to decode store creation", error: error)
+            }
+        }
     }
 
     // MARK: - Writing
 
     func write(resources: [EnrichedResource]) {
-        guard let scope = core?.scope(for: ResourcesFeature.self) else {
-            return
-        }
-        scope.eventWriteContext { _, recordWriter in
-            resources.forEach {
-                recordWriter.write(value: $0)
+        scope.eventWriteContext { [weak self] _, recordWriter in
+            let unknownResources = resources.filter { self?.knownIdentifiers.contains($0.identifier) == false }
+            for resource in unknownResources {
+                recordWriter.write(value: resource)
             }
+            self?.knownIdentifiers.formUnion(Set(unknownResources.map { $0.identifier }))
+        }
+    }
+
+    enum Constants {
+        static let knownResourcesKey = "known-resources"
+        static let storeCreationKey = "store-creation"
+    }
+}
+
+extension Data {
+    enum SerializationError: Error {
+        case invalidData
+    }
+
+    func asTimeInterval() throws -> TimeInterval {
+        var value: TimeInterval = 0
+        guard count >= MemoryLayout.size(ofValue: value) else {
+            throw SerializationError.invalidData
+        }
+        _ = Swift.withUnsafeMutableBytes(of: &value) {
+            copyBytes(to: $0)
         }
+        return value
+    }
+
+    func asKnownIdentifiers() throws -> Set<String>? {
+        return try JSONDecoder().decode(Set<String>.self, from: self)
+    }
+}
+
+extension TimeInterval {
+    func asData() -> Data {
+        return Swift.withUnsafeBytes(of: self) {
+            Data($0)
+        }
+    }
+}
+
+extension Set<String> {
+    func asData() -> Data? {
+        return try? JSONEncoder().encode(self) // Never fails
```

Comment:
suggestion/ Similar, `JSONEncoder` is a class so we should cache it instead of allocating memory each time.